shoes-core/lib/shoes/widget.rb

```diff
@@ -21,40 +21,41 @@ class Shoes
   #     say_hello 'Shoes'
   #   end
   #
-  class Widget
-    include Common::Inspect
-
+  class Widget < Shoes::Flow
     Shoes::App.subscribe_to_dsl_methods self
 
-    attr_accessor :parent
-    attr_writer :app
+    attr_accessor :original_args
 
-    class << self
-      attr_accessor :app
+    def handle_block(*_)
```

Comment:
just one thing, do we have redrawing etc. correctly setup for this?

Reply:
How do you mean? Because of where this is called it wires into the startup sequence correctly, and after that it's just a slot so everything's happy.